### DISCUSSION TITLE
Adding a plugin to a stage

### DISCUSSION THREAD

**Discussion by DomonkosSuranyi** [bevyengine/bevy#3858] [Q&A]
Hi all!
Is there any way to add a plugin to a stage? If there's no such way I'm curious to the reason why.

My concrete issue:
On the game server I most of my systems should run at a fixed framerate.
I've found one way to do so: Adding the systems to a custom stage where I set run criteria to FixedFramestep.
The issue is that there are Plugins too that I would like to run with exactly the same timestep but I couldn't find a way to add the plugin to my FixedUpdate stage.

**Comment by DomonkosSuranyi** [comment]
The same issue occurs when I want a plugin to run only in a specific state.
E.g. at "Connecting" state (where client tries to connect to server) I don't really need the CollisionPlugin to run...

**Comment by bjorn3** [comment]
There currently isn't. A plugin may currently add systems to any and all stages it wants. You may be interested in https://github.com/bevyengine/rfcs/pull/45.

**Comment by rlidwka** [comment]
> Is there any way to add a plugin to a stage? If there's no such way I'm curious to the reason why.

Ask plugin authors to add a configuration flag named `dont_register_your_systems_let_me_do_it_instead`.

Then register their stuff manually, here's an example how to do this with rapier3d:

https://github.com/dimforge/bevy_rapier/issues/343
